### PR TITLE
🌱 Add DockerMachinePool conversion

### LINKED ISSUE #5279: Add conversion for DockerMachinePool
Let's add conversion (impl, webhook, unit test) for DockerMachinePools.

/kind bug
[One or more /area label. See https://github.com/kubernetes-sigs/cluster-api/labels?q=area for the list of labels]

### PR DESCRIPTION
**Pull request by sbueringer**
Signed-off-by: Stefan Bueringer <buringerst@vmware.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #5279

### CODE REVIEW

File: test/infrastructure/docker/exp/api/v1beta1/dockermachinepool_webhook.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+func (r *DockerMachinePool) SetupWebhookWithManager(mgr ctrl.Manager) error {
```
Comment (sbueringer) [inline test/infrastructure/docker/exp/api/v1beta1/dockermachinepool_webhook.go:23]: In theory that's not necessary as the other webhooks already registery the conversion webhook, but that would be too implicit.

File: test/infrastructure/docker/main.go [hunk 2]
```diff
@@ -197,4 +197,9 @@ func setupWebhooks(mgr ctrl.Manager) {
 		setupLog.Error(err, "unable to create webhook", "webhook", "DockerClusterTemplate")
 		os.Exit(1)
 	}
+
+	if err := (&infraexpv1.DockerMachinePool{}).SetupWebhookWithManager(mgr); err != nil {
```
Comment (fabriziopandini) [inline test/infrastructure/docker/main.go:201]: This should be added only if `feature.Gates.Enabled(feature.MachinePool)`

**Review by fabriziopandini — COMMENTED** [review]
lgtm apart a small nit

**Comment by sbueringer** [comment]
@fabriziopandini Thx, fixed, PTAL

**Review by vincepri — APPROVED** [review]
/approve
/lgtm

**Pull request merged**